### PR TITLE
PR for installation-related changes

### PR DESCRIPTION
**Pull request by mosesyhc**

### CODE REVIEW

File: .flake8
Comment (mosesyhc) [inline .flake8:1]: Note: I don't understand why it has never been an issue until now. But the autogenerated _version.py file is now both ignored by git and flake8.

**Comment by wildsm** [comment]
What role do the empty `__init__.py` in the tests subdirs play?

**Comment by mosesyhc** [comment]
> What role do the empty `__init__.py` in the tests subdirs play?

The empty `__init__.py` originally is designed for discovery of the tests. However, pytest and our test script (that uses pytest) does not require them to discover the tests.

**Comment by mosesyhc** [comment]
Possible good resource for discussion: https://pytest.org/en/7.4.x/explanation/goodpractices.html

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,8 @@
 numpy>=1.18.3
 scipy>=1.7
-scikit-learn>=1.2.0
\ No newline at end of file
+scikit-learn>=1.2.0
+
+-e .
```
Comment (wildsm) [inline requirements.txt:5]: This was new to me

**Review by wildsm — COMMENTED** [review]
Approved (sorry, forgot to check off)

File: requirements.txt (hunk 1 above)
Comment (mosesyhc) [reply requirements.txt:5]: This identifies basic dependencies and developer dependencies.  Anything below `-e .` will only get checked if `-r requirements.txt` is explicitly invoked.

**Pull request merged**